docassemble/ALToolbox/data/questions/al_income.yml

```diff
@@ -725,6 +725,31 @@ subquestion: |
 fields:
   - Job name: x[i].source
 ---
+generic object: ALJobList
+continue button field: x.revisit
+question: |
+  Edit jobs
+subquestion: |
+  ${ x.table }
+
+  ${ x.add_action() }
+---
+generic object: ALJobList
+table: x.table
+rows: x
+columns:
+  - Title: |
+      row_item.source
+  - Employer: |
+      row_item.employer.name if hasattr(row_item.employer.name, "first") else ""
+  - Annual Income: |
+      currency(row_item.net_total())
+edit:
+  - source
+  - employer.name.first
```

Comment:
This works because `x.employer.name.first` is shown on the same screen as `x.is_self_employed`.  I think my thought here was that `is_self_employed` is an implementation detail; it's only used to make the user's life easier, but it's not information that goes on the form. `employer.name.first` is (`if x.is_self_employed == True`, employer.name is set to x.name).

Made a separate comment on #141, but it's because it needs to be declared in a different code block. You didn't mention what file you were using for testing, but you'll have to add a similar code block to define where the user's name is coming from.

Review screens are so messy that I'm tempted to say you just need to override this block if you change the other screens that collect information.

Reply:
I made a comment in the issue that I hope clarifies what I meant.

Reply:
This should resolve #141 now, I'll test that everything in the demo in #145 works as well